src/disco/containers/DiscoPane.js

```diff
@@ -152,21 +162,10 @@ export class DiscoPaneBase extends React.Component {
   }
 }
 
-function loadedAddons(state) {
-  return state.discoResults.map((result) => ({ ...result, ...state.addons[result.addon] }));
-}
-
-export function loadDataIfNeeded({ store: { dispatch, getState } }) {
-  const state = getState();
-  const addons = loadedAddons(state);
-  if (addons.length > 0) {
-    return Promise.resolve();
-  }
-  return getDiscoveryAddons({ api: state.api })
-    .then(({ entities, result }) => {
-      dispatch(loadEntities(entities));
-      dispatch(discoResults(result.results.map((r) => entities.discoResults[r])));
-    });
+export function loadedAddons(state) {
+  return state.discoResults.map(
+    (result) => ({ ...result, ...state.addons[result.addon] })
```

Comment:
but...

![white-walkers-what-will-happen-when-i-remove-the-spreads](https://user-images.githubusercontent.com/55398/29218454-7d024f5e-7e7a-11e7-8af2-aecdbf366291.jpg)


These are old spreads 😢 

They were just moved from: https://github.com/mozilla/addons-frontend/pull/2915/files#diff-0a5f326b61d0fe1d07f721f0d66cacaaL156